watertap/unit_models/selective_oil_permeation.py

```diff
@@ -0,0 +1,669 @@
+            self.flowsheet().config.time,
+            initialize=0.01,
+            bounds=(0.0, 1e3),
+            domain=NonNegativeReals,
+            units=units_meta("length") * units_meta("time") ** -1,
+            doc="Shell side liquid velocity at module inlet",
+        )
+
+        # Build control volume for feed side
+        self.feed_side = ControlVolume0DBlock(
+            dynamic=False,
+            has_holdup=False,
+            property_package=self.config.property_package,
+            property_package_args=self.config.property_package_args,
+        )
+
+        self.feed_side.add_state_blocks(has_phase_equilibrium=False)
+
+        self.feed_side.add_material_balances(
+            balance_type=self.config.material_balance_type, has_mass_transfer=True
+        )
+
+        self.feed_side.add_isothermal_assumption()
+
+        self.feed_side.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            has_pressure_change=self.config.has_pressure_change,
+        )
+
+        # Add permeate block
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = False  # permeate block is not an inlet
+        self.properties_permeate = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of permeate",
+            **tmp_dict,
+        )
+
+        # Add Ports
+        self.add_inlet_port(name="inlet", block=self.feed_side)
+        self.add_outlet_port(name="retentate", block=self.feed_side)
+        self.add_port(name="permeate", block=self.properties_permeate)
+
+        # References for control volume
+        # pressure change
+        if (
+            self.config.has_pressure_change is True
+            and self.config.momentum_balance_type != "none"
+        ):
+            self.deltaP = Reference(self.feed_side.deltaP)
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Average transmembrane pressure",
+        )
+        def eq_pressure_transmemb(b, t):
+            return (
+                b.pressure_transmemb[t]
+                == 0.5
+                * (
+                    b.feed_side.properties_in[t].pressure
+                    + b.feed_side.properties_out[t].pressure
+                )
+                - b.properties_permeate[t].pressure
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            doc="Mass transfer term",
+        )
+        def eq_mass_transfer_term(b, t, p, j):
+            if j == "H2O":
+                self.feed_side.mass_transfer_term[t, p, "H2O"].fix(
+                    0
+                )  # no water transfer
```

Comment:
seems very ideal to assume _no_ water transfer

Reply:
Probably, but that is what the UT folks have told us to assume. I think this is the assumption we want to stick with for now.